### DISCUSSION TITLE
multiple windows?

### DISCUSSION THREAD

**Discussion by xerocross** [Zettlr/Zettlr#3089] [Ideas]
Hi all. I'm a new user and new here to the repo. I wonder if there has been any talk of adding a feature so one can open a zettel/note in its own window. I didn't see anything like that in the issues section. Here's how I see this working: I click the flie tab in the main files pane and I drag it out into space somewhere else. Then it opens in its own window, subordinate to the main window (in the sense that closing the main window would still close the app). Thoughts? Has this idea already been brought up and sidelined or rejected?

**Comment by nathanlesage** [comment]
That is planned, and I'm almost finished with rewriting the code base in such a way that it becomes possible! Still gonna take some time, but it will come!

**Reply by xerocross** [reply]
yay! I'm a patron (on Patreon) and a Zettelkasten enthusiast. I'm very much in favor of this idea.

I'm also (at least formerly) a professional javascript engineer. (Currently, I do Java web backend.) I might like to contribute if I am able. I've never jumped in on an open-source project before.

**Reply by nathanlesage** [reply]
Uh, that's nice! If you ever want to contribute, just say so or open a PR, every help is very welcome :) And thank you for sponsoring!

**Reply by xerocross** [reply]
I wouldn't know where to begin contributing. That's maybe the main hurdle with people like me who might like to contribute. I would suggest that if open projects want to encourage participation then someone should act as a kind of team lead who can review a person's skill and point them to a bit of work that needs doing. Does this project have anyone like that?

**Reply by nathanlesage** [reply]
Yes, that would be me! I'm more than happy to show you around!

**Reply by xerocross** [reply]
I was formerly a professional JS developer. I worked on the LendingTree.com offers page for almost a year. That was a Vue.JS platform. I work as a Java backend dev now, but my JS skills are still there in the background. I have approx 3 years total dev exp. I have never worked professionally with TS before, but I find it easy to understand and I have played with it some in hobby work. I have never worked in Electron before.

Do you think there is anything I can contribute?

**Reply by nathanlesage** [reply]
Absolutely. The main process is basically a node server with some extra sugar added via Electron, but the renderers are fundamentally just websites, based on Vue. The only thing we currently have is the IPC system, but if you imagine IPC as being equal to some API that you can `fetch` stuff from, you're good to go!

**Reply by xerocross** [reply]
When I have some time and I'm in the right mood, I'll see if I can get the app to run from source on my computer. That sounds like a good place to start. Btw, as for the orginal topic of my post, I think it's one of the most essential items that Zettlr currently lacks. I know your time is your time and all that. Just consider my comment encouragement. :)

**Reply by nathanlesage** [reply]
Believe me — I use the app for my dissertation every day, and now, after four years, it really annoys me that I can't have multiple split editors and windows 😄  I have been working for the past years to gradually move the code into a shape that I will be able to do so in the coming two months or so! And yeah, thanks for any contribution you do, no matter how long it takes or how small — everything helps! :)

**Reply by xerocross** [reply]
I wish I had been familiar with the Zettlkasten method back when I was a PhD student. I'm not sure how helpful it would have been in my field. I studied math. But I bet it would have helped. Nowadays, I don't do math anymore. I'm a Java developer. But I enjoy using my Zettelkasten for its own sake. I write notes in it on a great variety of topics having nothing to do with my work.

I suppose there's no need then to explain why being able to open notes in multiple windows would be such a useful feature, so I won't generalize. I'll just say that in my case it would be very useful if I could hold one or two central notes open for curating in their own windows while I did research in the main window to make connections, updates, etc., to other notes.

**Reply by giz476234** [reply]
> That is planned, and I'm almost finished with rewriting the code base in such a way that it becomes possible! Still gonna take some time, but it will come!

@nathanlesage Very happy to see that you are already considering this! This would be a VERY useful feature!
Any estimation on when this would realistically be coming?

Thanks to the great improvements on the list behavior (e.g. images inside list items) in the recent releases, I'm can now almost move to Zettlr as my daily driver and I also plan to become a Patreon. Great project! Thanks for all the hard work!

**Reply by nathanlesage** [reply]
Still something to do, but we're now much closer than 3mo ago. Specifically, something I am currently working on is to improve the data for the editor, so that this is also prepared for that. But I can't give any ETA right now, since my PhD is very demanding -- but rest assured that I spend all of my available time working on it.

**Reply by giz476234** [reply]
Thanks, happy to hear that!
I could imagine that the following GUI mechanisms would be really useful in this context:
1. Right click on a folder/file in the file list on the left or a headline in the table of contents and then click on "Open in new Window". This would then ideally open a new complete Zettlr window (with all toolbars etc.) and navigate to that position. In this way you can immediately continue working there.
2. Right click on a tab at the top and then click on "Move to new Window" (similar to Firefox). Maybe later even being able to select to which Zettlr instance you want to move it to (in case you have several of them open).
3. Press a keyboard shortcut (or click a button in the GUI somehwere) to duplicate the current window with exactly the same position and open tabs, so you have the same view open twice (similar to Microsoft OneNote shortcut Ctrl + M)
It would be so amazing to have something like this.

Thanks again for all your effort and good luck for your PhD!

**Comment by Duckbilled** [comment]
Maybe it would be a feature request of it's own, but it would have been amazing if there was an option to keep te window on top of other applications. For example when you are taking notes of a website which is full screen and you can keep Zettlr in one of the corners.

<img width="1678" alt="Screenshot 2022-03-31 at 17 49 04" src="https://user-images.githubusercontent.com/52334713/161096813-3ac42cd1-ebf8-4bd6-a675-d4c86aebc72d.png">

**Reply by nathanlesage** [reply]
>  if there was an option to keep te window on top of other applications

I currently ponder the thought of adding a "scratchpad" functionality accessible from the Tray bar icon, which would then, in fact, stay on top.

**Reply by Duckbilled** [reply]
That sounds amazing, I wish I could help out, but have zero to none development skills. It is already an amazing app though, thanks you so much!

**Reply by Duckbilled** [reply]
You actually already have the icon in the menu bar (is that the one you meant, the top one, or do you mean the dock?), actually haven’t really used it yet. 

One could recreate that apple notes thingy a bit by setting a corner action, not sure if it supports opening a specific app.

**Reply by nathanlesage** [reply]
Yes I meant that thing here, which is as of now relatively uninspiring:

<img width="118" alt="image" src="https://user-images.githubusercontent.com/17251683/161389297-e7ac4ca0-84bc-4f49-9968-92b8cc7cd278.png">

**Comment by hajush** [comment]
Thank you so much for the Zettlr application, really love it. I've solved the issue in an ugly way by using another piece of software - Sublime. It would be so great to have multiple windows. Ideal would be to allow a split window. I'd prefer a vertical split - though having horizontal as well would be great). But just allowing two instances would suffice so I don't have to use a completely different user interface.

**Reply by nathanlesage** [reply]
I recommend you use the beta. That does exactly that.

**Reply by hajush** [reply]
Darn - I was excited to try this. The install worked okay, but it won't start. Should I file an issue?

A JavaScript error occurred in the main process
Uncaught Exception:
Error: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /opt/Zettlr/resources/app.asar.unpacked/.webpack/main/native_modules/build/Release/Nodehun.node)
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:2:1822)
    at Module._extensions..node (node:internal/modules/cjs/loader:1226:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:2:2049)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
    at Module._load (node:internal/modules/cjs/loader:846:12)
    at f._load (node:electron/js2c/asar_bundle:2:13328)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at 19572 (/opt/Zettlr/resources/app.asar/.webpack/main/index.js:2:1448444)
    at r (/opt/Zettlr/resources/app.asar/.webpack/main/index.js:2:3392547)
[946420:0219/075036.055440:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization

**Reply by nathanlesage** [reply]
That has been fixed by #4155; next Nightly (comes on Monday during the day) should fix that.
